Connections.md

```diff
@@ -0,0 +1,10 @@
+# Improved handling of failing connection attempts
+
+In versions prior to .NET 6, in case there is no connection immediately available in the connection pool, a new HTTP request will always spin up a new connection attempt and wait for it. The downside to this is, if it takes a while to establish that connection and another connection becomes available in the meantime, that request would continue waiting for the connection it spawned, hurting latency. In .NET 6.0 [we changed this](https://devblogs.microsoft.com/dotnet/dotnet-6-networking-improvements/#other-http-changes) to process requests on whichever connection that becomes available first, whether that’s a newly established one or one that became ready to handle the request in the meantime.
```

Comment:
I'd start with NET 6 behavior, since that's what caused the issues and what we're mitigating. Explaining 5.0 behavior is fine, but I wouldn't put it as the first thing.

Reply:
I think most readers are unaware of the changes in previous versions. It's very hard to clearly and unambiguously explain the story without mentioning the old behavior. To me it feels better to keep the historical context here.